ironic/drivers/modules/drac/inspect.py

```diff
@@ -35,16 +42,154 @@
-    should be implemented by this class.
-    """
-    pass
+    def inspect_hardware(self, task):
+        """Inspect hardware to get the hardware properties.
+
+        Inspects hardware to get the essential properties.
+        It fails if any of the essential properties
+        are not received from the node.
+
+        :param task: a TaskManager instance.
+        :raises: HardwareInspectionFailure if essential properties
+                 could not be retrieved successfully.
+        :returns: The resulting state of inspection.
+
+        """
+        super(DracRedfishInspect, self).inspect_hardware(task)
+
+        node = task.node
+        pxe_dev_nics = self._get_pxe_dev_nics(task)
+        if pxe_dev_nics is None:
+            LOG.warning("No PXE enabled NIC was found for node "
+                        "%(node_uuid)s.", {'node_uuid': node.uuid})
+
+        ports = objects.Port.list_by_node_id(task.context, node.id)
+        if ports:
+            for port in ports:
+                is_pxe_dev_nic = (port.address.upper() in pxe_dev_nics)
+                if port.pxe_enabled != is_pxe_dev_nic:
+                    port.pxe_enabled = is_pxe_dev_nic
+                    port.save()
+                    LOG.info('Port updated with pxe enabled %(pxe)s '
+                             'for node %(node_uuid)s during inspection',
+                             {'pxe': port.pxe_enabled, 'node_uuid': node.uuid})
+        else:
+            LOG.warning("No port information discovered "
+                        "for node %(node)s", {'node': node.uuid})
+
+        return states.MANAGEABLE
+
+    def _get_pxe_dev_nics(self, task):
+        """Get a list of pxe device interfaces.
+
+        :param task: a TaskManager instance.
+        :returns: Returns list of pxe device interfaces.
+        """
+        system = redfish_utils.get_system(task.node)
+
+        pxe_dev_nics = []
+        pxe_params = ["PxeDev1EnDis", "PxeDev2EnDis",
+                      "PxeDev3EnDis", "PxeDev4EnDis"]
+        pxe_nics = ["PxeDev1Interface", "PxeDev2Interface",
+                    "PxeDev3Interface", "PxeDev4Interface"]
+
+        redfish_ip = task.node.driver_info.get('redfish_address')
+        redfish_username = task.node.driver_info.get('redfish_username')
+        redfish_password = task.node.driver_info.get('redfish_password')
+
+        if system.boot.mode == 'uefi':
+            for param, nic in zip(pxe_params, pxe_nics):
+                if system.bios.attributes[param] == 'Enabled':
+                    nic_id = system.bios.attributes[nic]
+                    # Get MAC address of the given nic_id
+                    mac_response = requests.get('%s%s%s' % (
+                        redfish_ip, _MAC_URL, nic_id), auth=(
+                        redfish_username, redfish_password), verify=False)
+                    mac_address = json.loads(mac_response.__dict__.get(
+                        '_content')).get('MACAddress')
```

Comment:
It would be great if you could create a _get_mac_address() method and call it in the 2 places in this method.

Reply:
Done